src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -131,54 +196,54 @@ protected final Checker createChecker(Configuration moduleConfig)
     }
 
     /**
-     * Creates {@link Checker} instance based on specified {@link Configuration}.
+     * Creates {@link Checker} instance based on the given {@link Configuration} instance.
      *
      * @param moduleConfig {@link Configuration} instance.
      * @param moduleCreationOption {@code IN_TREEWALKER} if the {@code moduleConfig} should be added
      *                                                  under {@link TreeWalker}.
-     * @return {@link Checker} instance.
+     * @return {@link Checker} instance based on the given {@link Configuration} instance.
      * @throws Exception if an exception occurs during checker configuration.
      */
     protected final Checker createChecker(Configuration moduleConfig,
-                                    ModuleCreationOption moduleCreationOption)
+                                 ModuleCreationOption moduleCreationOption)
             throws Exception {
-        final Configuration dc;
+        final Checker checker = new Checker();
+        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+        // make sure the tests always run with English error messages
+        // so the tests don't fail in supported locales like German
+        final Locale locale = Locale.ENGLISH;
+        checker.setLocaleCountry(locale.getCountry());
+        checker.setLocaleLanguage(locale.getLanguage());
 
         if (moduleCreationOption == ModuleCreationOption.IN_TREEWALKER) {
-            dc = createTreeWalkerConfig(moduleConfig);
+            final Configuration dc = createTreeWalkerConfig(moduleConfig);
```

Comment:
Can we improve naming while we refactor? `dc` is just `defaultConfiguration` right?

Reply:
Done.

Reply:
Resolved.